msmbuilder/featurizer/multichain.py

```diff
@@ -0,0 +1,510 @@
+            self.align_indices = align_indices
+        else:
+            self.align_indices = self._get_atom_range(self.reference_frame,
+                                                 self.align_by)
+
+        if calculate_indices is not None:
+            if calculate_for is not 'custom':
+                if not self._check_indices(self.reference_frame,
+                                           self.calculate_for,
+                                           calculate_indices):
+                    raise ValueError("calculate_indices must be on the " \
+                                     "calculate_for chain")
+            else:
+                if not all(calculate_indices[i] in
+                           range(self.reference_frame.n_atoms)
+                           for i in range(len(calculate_indices))):
+                    raise ValueError("calculate_indices must exist")
+            self.calculate_indices = calculate_indices
+        else:
+            self.calculate_indices = self._get_atom_range(self.reference_frame,
+                                                     self.calculate_for)
+
+
+    # custom option will never see this
+    def _get_atom_range(self, traj, chain):
+        if chain == self.protein_chain:
+            offset = self.p_atom_offset
+        elif chain == self.ligand_chain:
+            offset = self.l_atom_offset
+        else:
+            raise ValueError("Protein or ligand chain required")
+        atom_range = range(offset,offset+traj.top.chain(chain).n_atoms)
+        return atom_range
+
+
+    # custom option will never see this
+    def _check_indices(self, traj, chain, indices):
+        atom_range = self._get_atom_range(traj, chain)
+        return all(indices[i] in atom_range for i in range(len(indices)))
+
+
+    def _naive_rmsd(self, traj, ref, idx):
+        return np.sqrt(np.sum(np.square(traj.xyz[:,idx,:] - ref.xyz[:,idx,:]),
+                              axis=(1, 2))/len(idx))
+
+
+    def partial_transform(self, traj):
+        """Featurize an MD trajectory into a vector space via distance
+        after superposition
+
+        Parameters
+        ----------
+        traj : mdtraj.Trajectory
+            A molecular dynamics trajectory to featurize.
+
+        Returns
+        -------
+        features : np.ndarray, shape=(n_frames, n_ref_frames)
+            The RMSD value of each frame of the input trajectory to be
+            featurized versus each frame in the reference trajectory. The
+            number of features is the number of reference frames.
+
+        See Also
+        --------
+        transform : simultaneously featurize a collection of MD trajectories
+        """
+        X = np.zeros((traj.n_frames, self.n_features))
+
+        for f in range(self.n_features):
+            frame = self.reference_traj[f]
+            traj.superpose(frame, atom_indices=self.align_indices)
+            X[:,f] = self._naive_rmsd(traj, frame, self.calculate_indices)
```

Comment:
I'm pretty sure I tried that and it didn't work. I can look again, and if it doesn't work again I'll follow up.